newm-server/src/main/kotlin/io/newm/server/features/minting/MintingMessageReceiver.kt

```diff
@@ -124,14 +127,32 @@ class MintingMessageReceiver : SqsMessageReceiver {
                     songId = mintingStatusSqsMessage.songId,
                     mintingStatus = MintingStatus.AwaitingAudioEncoding,
                 )
+                mintingStatusHistoryRepository.add(
```

Comment:
Instead of putting these throughout this code, move it into `songRepository.updateSongMintingStatus()`. That way, you won't miss any updates.